tests/base.py

```diff
@@ -222,14 +223,27 @@ def modify_expected_datatypes(self, expected_records):
             for record in expected_records:
                 for key, value in record.items():
                     self.align_date_type(record, key, value)
+                    self.sort_array_type(record, key, value)
             return
 
         for key, value in expected_records.items(): # Modify a single record
             self.align_date_type(expected_records, key, value)
+            self.sort_array_type(expected_records, key, value)
 
     def align_date_type(self, record, key, value):
         """datetime values must conform to ISO-8601 or they will be rejected by the gate"""
         if isinstance(value, str) and key in ['updated_at']:
             raw_date = self.parse_date(value)
             iso_date = dt.strftime(raw_date,  "%Y-%m-%dT%H:%M:%S.%fZ")
             record[key] = iso_date
+
+    def sort_array_type(self, record, key, value):
+        try:
+            if isinstance(value, list) and value and key in ['ads']:
+                if isinstance(value[0], dict) and "id" in value[0].keys():
```

Comment:
You sure you're only supposed to this for the first value? And not for every value? Cause in the next line you sort the value for that key. And if id was not in a later value (which I realize doesn't make much sense but could be possible) it might fail.

Reply:
That's a good point, I didn't write this one so I am not sure what the reason was for looking at the first value only. Once I have data for a list I can look into this more.